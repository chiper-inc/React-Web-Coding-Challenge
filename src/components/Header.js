import React from 'react';
import LogoApp from './images/logo-app.png';

export const Header = () => {

    return (
        <header>
            <h1>
                <img src={LogoApp} alt={LogoApp} />
            </h1>
            <h2>Police Department Berlin</h2>
            <h4>Stolen Bykes</h4>
        </header>
    )
}
