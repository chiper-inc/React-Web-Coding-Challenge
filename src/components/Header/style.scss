.header {
  @apply flex flex-col md:flex-row items-center justify-center p-4;
  background-color: #2600f8;
  box-shadow: 0px 8px 15px 3px #999;


  .content-img {
    width: 7.5rem;
    height: 7.5rem;
    img {
      width: 100%;
      height: 100%;
      object-fit: contain;
    }
  }

  .content-info {
    @apply flex flex-col justify-center md:ml-8;

    .title {
      @apply text-2xl md:text-4xl text-white uppercase;
    }
    .sub-title {
      @apply text-xl md:text-2xl text-yellow-400;

    }
  }
}
