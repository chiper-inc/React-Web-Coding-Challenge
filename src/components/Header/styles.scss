.Header {
  width: 80%;
  height: 100px;
  margin-left: auto;
  margin-right: auto;
  margin-top: 30px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  
  img {
    width: 60px;
    height: 80px;
    margin: auto;
  }

  div {
    width: 90%;
    height: 80px;
    margin: auto;
    h1 {
      margin: 0;
    }
    h4  {
      margin: 0
    }
  }
}
