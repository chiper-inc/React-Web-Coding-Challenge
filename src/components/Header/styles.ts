import { Box, styled } from '@mui/material';

export const MainHeader = styled(Box)`
  display: flex;

  > section:first-of-type {
    width: 100px;

    img {
      width: 100%;
    }
  }
`;
