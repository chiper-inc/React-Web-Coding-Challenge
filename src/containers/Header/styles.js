import styled from 'styled-components'

export const Wrapper = styled.nav`
  display: grid;
  grid-auto-flow: column;
  grid-template-columns: auto 1fr;
  grid-gap: 6px;
  padding: 20px 8%;
  background-color: #1a1a1a;
  color: #b5b5b5;
  box-shadow: 0px 2px 5px 0px rgb(0 0 0 / 20%);
  align-items: center;
`;

export const Text = styled.span`
  font-size: 1.4rem;
  font-weight: 600;
  color: #b5b5b5;
  text-align: center;
`;
