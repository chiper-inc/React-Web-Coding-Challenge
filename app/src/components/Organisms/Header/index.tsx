import { Forms, TitlePage } from '@/components/Molecules';
import React from 'react';

const Header: React.FC = () => {
  return (
    <>
      <TitlePage title="Police Department of Berlin" subtitle="Stolen Bykes" />
      <Forms />
    </>
  );
};

export { Header };
