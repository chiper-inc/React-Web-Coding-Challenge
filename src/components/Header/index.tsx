import React from 'react';
import { MainHeader } from './styles';
import logo from '../../assets/images/police-128.png';

function Header () {
  return (
        <MainHeader>
            <section>
                <img src={logo} alt=""/>
            </section>
            <section>
                <h1>Police Departament of Berlin</h1>
                <h3>Stolen bykes</h3>
            </section>
        </MainHeader>
  );
}

export { Header };
