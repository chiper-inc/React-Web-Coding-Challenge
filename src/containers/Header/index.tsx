import { BtnBack } from "../../components/BtnBack";
import { Wrapper, Text } from "./styles";

interface IProps {
  title?: string
}

export const Header = ({ title }: IProps) => {
  return (
    <Wrapper>
      <BtnBack/>
      <Text>{title}</Text>
    </Wrapper>
  );
};
