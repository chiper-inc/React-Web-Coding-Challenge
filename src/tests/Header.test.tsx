import React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header';
test('renders the component Header', () => {
  const component = shallow(<Header />);
  expect(component).toMatchSnapshot();
});
