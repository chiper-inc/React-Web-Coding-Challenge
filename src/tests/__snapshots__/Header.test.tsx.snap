// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="container mx-auto px-4 mt-8"
>
  <BrowserRouter>
    <Header />
    <Switch>
      <Route
        path="/details/:id"
      >
        <Details />
      </Route>
      <Route
        path="/"
      >
        <Home />
      </Route>
    </Switch>
  </BrowserRouter>
</div>
`;

exports[`renders the component Header 1`] = `
<div
  className="header"
>
  <div
    className="content-img"
  >
    <img
      alt="Logo"
      src="logo.jpg"
    />
  </div>
  <div
    className="content-info"
  >
    <Link
      className="title"
      to="/"
    >
      Police Department of Berlin
    </Link>
    <h4
      className="sub-title"
    >
      Stolen bykes.
    </h4>
  </div>
</div>
`;

exports[`renders the component Headerw 1`] = `
<div
  className="header"
>
  <div
    className="content-img"
  >
    <img
      alt="Logo"
      src="logo.jpg"
    />
  </div>
  <div
    className="content-info"
  >
    <Link
      className="title"
      to="/"
    >
      Police Department of Berlin
    </Link>
    <h4
      className="sub-title"
    >
      Stolen bykes.
    </h4>
  </div>
</div>
`;
